Orange/widgets/visualize/owviolinplot.py

```diff
@@ -0,0 +1,1092 @@
+        data = Input("Data", Table)
+
+    class Outputs:
+        selected_data = Output("Selected Data", Table, default=True)
+        annotated_data = Output(ANNOTATED_DATA_SIGNAL_NAME, Table)
+
+    class Error(OWWidget.Error):
+        no_cont_features = Msg("Plotting requires a numeric feature.")
+        not_enough_instances = Msg("Plotting requires at least two instances.")
+
+    KERNELS = ["gaussian", "epanechnikov", "linear"]
+    KERNEL_LABELS = ["Normal", "Epanechnikov", "Linear"]
+    SCALE_LABELS = ["Area", "Count", "Width"]
+
+    settingsHandler = DomainContextHandler()
+    value_var = ContextSetting(None)
+    order_by_importance = Setting(False)
+    group_var = ContextSetting(None)
+    order_grouping_by_importance = Setting(False)
+    show_box_plot = Setting(True)
+    show_strip_plot = Setting(False)
+    show_rug_plot = Setting(False)
+    order_violins = Setting(False)
+    orientation_index = Setting(1)  # Vertical
+    kernel_index = Setting(0)  # Normal kernel
+    scale_index = Setting(AREA)
+    selection_ranges = Setting([], schema_only=True)
+    visual_settings = Setting({}, schema_only=True)
+
+    graph_name = "graph.plotItem"
+    buttons_area_orientation = None
+
+    def __init__(self):
+        super().__init__()
+        self.data: Optional[Table] = None
+        self.orig_data: Optional[Table] = None
+        self.graph: ViolinPlot = None
+        self._value_var_model: VariableListModel = None
+        self._group_var_model: VariableListModel = None
+        self._value_var_view: ListViewSearch = None
+        self._group_var_view: ListViewSearch = None
+        self._order_violins_cb: QCheckBox = None
+        self._scale_combo: QComboBox = None
+        self.selection = []
+        self.__pending_selection: List = self.selection_ranges
+
+        self.setup_gui()
+        VisualSettingsDialog(
+            self, self.graph.parameter_setter.initial_settings
+        )
+
+    def setup_gui(self):
+        self._add_graph()
+        self._add_controls()
+
+    def _add_graph(self):
+        box = gui.vBox(self.mainArea)
+        self.graph = ViolinPlot(self, self.kernel,
+                                self.scale_index, self.orientation,
+                                self.show_box_plot, self.show_strip_plot,
+                                self.show_rug_plot, self.order_violins)
+        self.graph.selection_changed.connect(self.__selection_changed)
+        box.layout().addWidget(self.graph)
+
+    def __selection_changed(self, indices: List, ranges: List):
+        self.selection_ranges = ranges
+        if self.selection != indices:
+            self.selection = indices
+            self.commit()
+
+    def _add_controls(self):
+        self._value_var_model = VariableListModel()
+        sorted_model = SortProxyModel(sortRole=Qt.UserRole)
+        sorted_model.setSourceModel(self._value_var_model)
+        sorted_model.sort(0)
+
+        view = self._value_var_view = ListViewSearch()
```

Comment:
These ListViewSerach boxes need a greater minimum height. Now, these selectors are useless if the widget is small enough that the controlArea scrollbar is shown. 

![image](https://user-images.githubusercontent.com/552182/108349004-c2391300-71e2-11eb-8cae-625d30ef446f.png)

Why are they so small?